consensus/istanbul/backend/snapshot.go

```diff
@@ -178,84 +178,191 @@ func (s *Snapshot) apply(headers []*types.Header) (*Snapshot, error) {
 	snap := s.copy()
 
 	for _, header := range headers {
-		// Remove any votes on checkpoint blocks
-		number := header.Number.Uint64()
-		if number%s.Epoch == 0 {
-			snap.Votes = nil
-			snap.Tally = make(map[common.Address]Tally)
-		}
-		// Resolve the authorization key and check against validators
-		validator, err := ecrecover(header)
-		if err != nil {
-			return nil, err
-		}
-		if _, v := snap.ValSet.GetByAddress(validator); v == nil {
-			return nil, errUnauthorized
+		if isQBFTConsensus && header.Number.Uint64() > qbftBlockNumber {
```

Comment:
why do we need `header.Number.Uint64() > qbftBlockNumber`? `isQBFTConsensus ` should already handle that case

Reply:
`Snapshot.go::apply()` processes headers from previous block. The check here is to ensure if qbft consensus was enabled for that particular block height identified by the header